proposals/gc/MVP.md

```diff
@@ -565,3 +750,215 @@ See [GC JS API document](MVP-JS.md) .
 * Provide functionality to generate fresh, non-canonical RTTs?
 
 * Provide a way to make data types non-eq, especially immutable ones?
+
+
+
+## Appendix: Formal Rules
+
+### Validity
+
+#### Type Indices (`C |- <typeidx> ok`)
+
+```
+C(x) = ct
+---------
+C |- x ok
+```
+
+#### Value Types (`C |- <valtype> ok`)
+
+```
+
+-----------
+C |- i32 ok
+
+C |- x ok
+-------------
+C |- ref x ok
+```
+
+...and so on.
+
+
+#### Structural Types (`C |- <strtype> ok`)
+```
+(C |- t1 ok)*
+(C |- t2 ok)*
+--------------------
+C |- func t1* t2* ok
+
+(C |- ft ok)*
+------------------
+C |- struct ft* ok
+
+C |- ft ok
+----------------
+C |- array ft ok
+```
+
+#### Defined Types (`C |- <subtype>* ok(x)`)
+
+```
+C |- st ok
+(C |- st <: expand(C(x)))*
+(x < x')*
+--------------------------
+C |- sub x* st ok(x')
+
+C |- dt ok(x)
+C |- dt'* ok(x+1)
+-------------------
+C |- dt dt'* ok(x)
+```
+
+#### Type Definitions (`C |- <deftype>* -| C'`)
```

Comment:
I'm not familiar with this reverse turnstile syntax. What does it mean?

Reply:
Oh, actually, that should be `st` and `st*`. Will fix.